### DISCUSSION TITLE
Validator Errors

### DISCUSSION THREAD

**Discussion by AJLgeHG** [CMSgov/price-transparency-guide#518] [Q&A] [answered]
Is there a document available on validator errors that developers can reference when they run the machine-readable files through the validator and an error is returned? 

We're seeing the below error for our allowed-amounts file. We searched the files to find the offending data and have not seen any thing unusual, but it is a large file and we could be missing the problem. Does anyone know how to interpret this error and what could be causing the issue? 

@shaselton-usds  Is there a document developers can reference to interpret validator errors?

Input is not a valid JSON
Error(offset 0): Invalid value.

**The error report has:**
Input JSON is invalid.
Error Name: type
Message: Property has a type 'integer' that is not in the following list: 'string'.
Instance: #/out_of_network/0/allowed_amounts/0/service_code/3
Schema: #/definitions/allowed_amounts/properties/service_code/items

Invalid schema: #/definitions/allowed_amounts/properties/service_code/items
Invalid keyword: type
Invalid code: 20
Invalid message: Property has a type '%actual' that is not in the following list: '%expected'.
Invalid document: #/out_of_network/0/allowed_amounts/0/service_code/3
Error report:
{
    "type": {
        "expected": [
            "string"
        ],
        "actual": "integer",
        "errorCode": 20,
        "instanceRef": "#/out_of_network/0/allowed_amounts/0/service_code/3",
        "schemaRef": "#/definitions/allowed_amounts/properties/service_code/items"
    }
}

Thanks for your help.

**Comment by shaselton-usds** [comment, marked as answer]
@AJLgeHG Thanks for reaching out.

So, the key to locating where the error is on this line:
```
Instance: #/out_of_network/0/allowed_amounts/0/service_code/3
```
With the actual error being:
```
Message: Property has a type 'integer' that is not in the following list: 'string'.
```

This is somewhat of an XPath location of where the error is located.

 In this case, the error can be found at:
The first out_of_network object. #**/out_of_network/0**/allowed_amounts/0/service_code/3
The first allowed_amounts object. #/out_of_network/0/**allowed_amounts/0**/service_code/3
The fourth service_code value (in the array). #/out_of_network/0/allowed_amounts/0/**service_code/3**

So, in your first `out_of_network` object, first `allowed_amounts` object, fourth `service_code` value is an integer and it needs to be a string.
